presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -314,18 +327,47 @@ private TableHandle getTableHandle(ShowStats node, QualifiedName table)
             ImmutableList.Builder<String> columnNamesBuilder = ImmutableList.builder();
```

Comment:
It would probably be slightly better to create column names earlier and provide whole thing to both variants of createTableStatsRow to make sure that all data are organised as expected.